### PR TITLE
Add performance test nodes, using complex meshes.

### PR DESCRIPTION
**Pull request by doxygen-spammer**
```
Add performance test nodes, using complex meshes.

These are two types of marbles, which use meshes with many polygons.
One looks like glass, being mostly transparent.
The other shall look like metal with specular.
The specular is implemented as overlay tile with static color "white".

Both nodes are provided with "clip" and "blend" transparency versions.
The "blend" metal marble shows performance problem #12900.

These media files are a donation from the (unreleased) doxy_plush mod,
and are provided with CC0 permissions.
```

These 4 nodes are usable to reproduce #12900 easily.
I think that such “performance test nodes” are also likely useful for other performance related tests.

## To do

This PR is Ready for Review.

## How to test

Start devtest game and place these nodes.
(Search for “performance” in the Chest of Everything.)
The metal marbles can be colored with the param2 tool.

screenshot_20230120_125538

### CODE REVIEW

File: games/devtest/mods/testnodes/performance_test_nodes.lua [hunk 1]
```diff
@@ -0,0 +1,59 @@
+-- Performance test mesh nodes
+
+local S = minetest.get_translator("testnodes")
+
+-- Complex mesh
+minetest.register_node("testnodes:performance_mesh_clip", {
+	description = S("Performance Test Node\nMarble with 'clip' transparency"),
```
Comment (TurkeyMcMac) [inline games/devtest/mods/testnodes/performance_test_nodes.lua:7]: Descriptions should be translated like so: https://github.com/minetest/minetest/blob/cded6a3945206e51c35adbd063f1aec3a47e310f/games/devtest/mods/testnodes/param2.lua#L6-L7

**Comment by oong819** [comment]
dont let this die

**Comment by SmallJoker** [comment]
This PR does not have the necessary 2 approvals, hence I assume it was closed due to inactivity of the pending review. There should've been a note as for why it was closed.

I think this PR is okay but should apply pngcrush/optipng on the textures and also properly license all media.

**Comment by rubenwardy** [comment]
> When Zughy silently closed this in March, I assumed they merged it manually. So there is nothing left to do here. Am I wrong?

You didn't respond to the review left by TurkeyMcMac, so the PR was closed

**Comment by doxygen-spammer** [comment]
> You didn't respond to the review left by TurkeyMcMac, so the PR was closed

Thank you Rubenwardy, this resolves my confusion.

In fact I responded within just 6 hours via email, but only now I notice that Github didn’t process the email.
I will respond with the web interface right now.

File: games/devtest/mods/testnodes/performance_test_nodes.lua (hunk 1 above)
Comment (rubenwardy) [reply games/devtest/mods/testnodes/performance_test_nodes.lua:7]: Concatenating inside sentences is bad, but it's a good idea to concatenate multiple senstences together.

To rephrase, here's how the modding book puts it:

image

Comment (doxygen-spammer) [reply games/devtest/mods/testnodes/performance_test_nodes.lua]: I can probably answer it myself: There is not any example in devtest of translated line breaks, according to `grep -P 'S\("[^"]*\\n'`.
I changed it as requested.

**Comment by doxygen-spammer** [comment]
Now I understand.
It contains this section:

> Authors of media files
> -----------------------
> Everything not listed in here:
> Copyright (C) 2010-2012 celeron55
> 
> ShadowNinja:
>   textures/base/pack/smoke_puff.png
> [...]

So I think I may just “donate” everything to celeron55.

**Review by SmallJoker — APPROVED** [review]
An reproduce the performance issue.

* No palette (40 nodes): 800 FPS
* Palette + clip (10 nodes): 650 FPS
* Palette + blend (10 nodes): 200 FPS <-- bottleneck

This makes me wonder whether palette colors are evaluated for each pixel that is not fully transparent.

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**